File: patient/src/main/java/com/openspace/PatientManagement/service/SchoolService.java
package com.openspace.PatientManagement.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.openspace.Model.DoctorManagement.School;

public interface SchoolService {

	void addSchool(School school);

	List<School> getAllSchools();

	void updateSchool(School school);

	Page<School> getAllSchoolsByAdminPage(String username, int page, int size);

	List<School> getAllSchoolsByAdmin(String username);

	void deleteSchool(Long id);

	List<School> getAllSchoolsByCompany(String username);

}

File: portal/src/main/resources/static/service/school/schoolService.js
function schoolService(PATIENT_MODULE_CONFIG, $http) {
	this.addSchool = function(school) {
	return 	$http.post(PATIENT_MODULE_CONFIG.ADD_SCHOOL(),school);
	};
	/*this.getAllschools=function(){
		return $http.get(MODULE_CONFIG.GETALLSCHOOLS());
	};
	
	this.getAllschoolsByCompanyId=function(){
		return $http.get(MODULE_CONFIG.GETALLSCHOOLSBYCOMPANYID(id));
	};*/
	this.updateSchool=function(school){
		return $http.post(PATIENT_MODULE_CONFIG.UPDATE_SCHOOL(),school);
	};
	this.deleteSchool=function(id){
		return $http.delete(PATIENT_MODULE_CONFIG.DELETE_SCHOOL(id));
	};
	
	
	//GET
	this.getAllSchoolsByCompanyUserNamePage=function(companyAdminUsername,page,size){
		return $http.get(PATIENT_MODULE_CONFIG.GET_ALL_SCHOOLS_BY_COMPANY_USERNAME_PAGE(companyAdminUsername,page,size));
	};
	
	this.getAllSchoolsByCompany=function(companyAdminUsername){
		return $http.get(PATIENT_MODULE_CONFIG.GET_ALL_SCHOOLS_BY_COMPANY(companyAdminUsername));
	};
	
	this.getAllSchoolsByCompanyUsername=function(companyAdminUsername){
		return $http.get(PATIENT_MODULE_CONFIG.GET_ALL_SCHOOLS_BY_COMPANY_USERNAME(companyAdminUsername));
	};
	
}

angular.module('HealthApplication').service('schoolService', schoolService);





